scripts/xtensa-build-zephyr.py

```diff
@@ -302,22 +284,60 @@ def check_west_installation():
 			"https://docs.zephyrproject.org/latest/getting_started/index.html")
 	print(f"Found west: {west_path}")
 
+def west_reinitialize(west_root_dir: pathlib.Path, west_manifest_path: pathlib.Path):
+	"""[summary] Performs west reinitialization to SOF manifest file asking user for permission.
+	Prints error message if script is running in non-interactive mode.
+
+	:param west_root_dir: directory where is initialized.
+	:type west_root_dir: pathlib.Path
+	:param west_manifest_path: manifest file to which west is initialized.
+	:type west_manifest_path: pathlib.Path
+	:raises RuntimeError: Raised when west is initialized to wrong manifest file
+	(not SOFs manifest) and script is running in non-interactive mode.
+	"""
+	global west_top
+	message = "West is initialized to manifest other than SOFs!\n"
+	message +=  f"Initialized to manifest: {west_manifest_path}." + "\n"
+	dot_west_directory  = pathlib.Path(west_root_dir.resolve(), ".west")
+	if args.no_interactive:
+		message += f"Try deleting {dot_west_directory } directory and rerun this script."
+		raise RuntimeError(message)
+	question = message + "Reinitialize west to SOF manifest? [Y/n] "
+	print(f"{question}")
+	while True:
+		try:
+			reinitialize_answer = strtobool(input().lower())
+			break
+		except ValueError:
+			sys.stdout.write('Please respond with \'Y\' or \'n\'.\n')
+	if not reinitialize_answer:
+		print("Can not proceed. Reinitialize your west manifest to SOF and rerun this script.")
+		exit(-1)
+	shutil.rmtree(str(dot_west_directory ), ignore_errors=True)
+	execute_command(["west", "init", "-l", f"{SOF_TOP}"], cwd=west_top)
+
 def find_west_workspace():
-	"""[summary] Scans for west workspaces using west topdir command by checking script execution
-	west_top and SOF_TOP directories.
+	"""[summary] Validates whether west workspace had been initialized and points to SOF manifest.
 
-	:return: [description] Returns path when west workspace had been found, otherwise None
-	:rtype: [type] bool, pathlib.Path
+	:return: [description] Returns west topdir and path to west manifest west.yml, otherwise None, None.
+	:rtype: [type] pathlib.Path, pathlib.Path
 	"""
 	global west_top, SOF_TOP
-	paths = [ pathlib.Path(os.getcwd()), SOF_TOP, west_top]
-	for path in paths:
-		if path.is_dir():
-			result = execute_command(["west", "topdir"], capture_output=True,
-						 text=True, check=False, cwd=path)
-		if result.returncode == 0:
-			return pathlib.Path(result.stdout.rstrip(os.linesep))
-	return None
+	west_manifest_path = pathlib.Path(SOF_TOP, "west.yml")
+	result_rootdir = execute_command(["west", "topdir"], capture_output=True, cwd=west_top,
+		timeout=10, check=False)
+	if result_rootdir.returncode != 0:
```

Comment:
Well, yes but no in this case.
Look at the check flag docs fragment

> If check is true, and the process exits with a non-zero exit code, a [CalledProcessError](https://docs.python.org/3/library/subprocess.html#subprocess.CalledProcessError) exception will be raised. Attributes of that exception hold the arguments, the exit code, and stdout and stderr if they were captured.

So its not like we catch an exception from subprocess directly but python command [subprocess.run](https://docs.python.org/3/library/subprocess.html#subprocess.run) generates that exception and raises it if you set check=True. So in this case asking subprocess.run to raise exception just to catch it seems irrational.

Reply:
The operating system only provides error codes, so all `subprocess` exceptions are "faked" anyway. subprocess synthesizes exceptions because it's more "pythonic".

My suggestion to catch `CalledProcessError` was also a suggestion to let other, unexpected exceptions/error codes "crash" the script with a stack trace to help debug it.

Nevermind, no real issue here.